attestation-service/verifier/src/lib.rs

```diff
@@ -89,12 +89,28 @@ pub type TeeEvidenceParsedClaim = serde_json::Value;
 
 #[async_trait]
 pub trait Verifier {
-    /// Verify the hardware signature and report data in TEE quote.
-    /// If the verification is successful, a key-value pairs map of TCB status will be returned,
-    /// The policy engine of AS will carry out the verification of TCB status.
+    /// Verify the hardware signature.
+    ///
+    ///
+    /// If `report_data` is given, the binding of the `report_data`
+    /// against the `report_data` inside the hardware evidence will
+    /// be checked. So do `init_data_hash`.
+    ///
+    ///
+    /// Semantically, a `report_data` is a byte slice given when
+    /// a hardware evidence is generated. The `report_data` will be
+    /// included inside the hardware evidence, thus its integrity will
+    /// be protected by the signature of the hardware.
+    ///
+    ///
+    /// A `init_data_hash` is another byte slice given when the TEE
+    /// instance is created. It is always provided by untrusted host,
+    /// but its integrity will be protected by the tee evidence.
+    /// Typical `init_data_hash` is `HOSTDATA` for SNP.
     async fn evaluate(
         &self,
-        nonce: String,
-        attestation: &Attestation,
+        evidence: &[u8],
+        report_data: Option<&[u8]>,
```

Comment:
Would it be clearer to call this `expected_report_data` and `expected_init_data_hash`?

Reply:
nice idea. Let me fix